addons/pos_self_order/static/src/Pages/ProductMainView/ProductMainView.js

```diff
@@ -0,0 +1,91 @@
+/** @odoo-module */
+
+import { Component, useState } from "@odoo/owl";
+import { useSelfOrder } from "@pos_self_order/SelfOrderService";
+import { NavBar } from "@pos_self_order/Components/NavBar/NavBar";
+import { FloatingButton } from "@pos_self_order/Components/FloatingButton/FloatingButton";
+import { IncrementCounter } from "@pos_self_order/Components/IncrementCounter/IncrementCounter";
+export class ProductMainView extends Component {
+    static template = "pos_self_order.ProductMainView";
+    static props = { product: Object };
+    static components = {
+        NavBar,
+        FloatingButton,
+        IncrementCounter,
+    };
+    setup() {
+        this.selfOrder = useSelfOrder();
+
+        // we want to keep track of the last product that was viewed
+        this.selfOrder.currentProduct = this.props.product.product_id;
+
+        this.privateState = useState({
```

Comment:
```suggestion
        this.state = useState({
```

All fields and methods in a component are implicitly private.